lib/mayaUsd/ufe/ProxyShapeSceneSegmentHandler.cpp

```diff
@@ -118,5 +121,13 @@ bool ProxyShapeSceneSegmentHandler::isGateway_(const Ufe::Path& path) const
                  : _mayaSceneSegmentHandler ? _mayaSceneSegmentHandler->isGateway_(path) : false;
 }
 
+Ufe::Path ProxyShapeSceneSegmentHandler::rootSceneSegmentRootPath() const
+{
+    const auto*       pyCommand = "import maya.api.OpenMaya as om; it = om.MItDag(); root = "
+                                  "it.currentItem(); om.MFnDependencyNode(root).name();";
+    const std::string mayaRootPath = MGlobal::executePythonCommandStringResult(pyCommand).asChar();
+    return Ufe::PathString::path(mayaRootPath);
```

Comment:
I like the idea of having a Maya SceneSegmentHandler registered by Maya.  However, this introduces a Maya version dependency (e.g. this functionality is present only in Maya 2026 and up).

Reply:
Hm, but adding the `rootSceneSegmentRootPath()` method in UFE is a breaking change, isn't it? So, it will only be available in Maya 2026 and up anyway, no? I don't think we planned on backporting it to Maya 2025.

Reply:
Two small things about this code. It should be pure C++ (not calling Python) and once you have the root node name you should create the path directly using Ufe::Path (not thru pathstring). Something like this:

```
#include <maya/MItDag.h>
#include <maya/MFnDependencyNode.h>

MItDag it;
auto              root = it.currentItem();
MFnDependencyNode rootNode(root);
std::string       rootName(rootNode.name().asChar());
Ufe::Path         rootPath(Ufe::PathSegment(rootName, MayaUsd::ufe::getMayaRunTimeId(), '|'));
return rootPath;
```